iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/task/CrossSpaceCompactionTask.java

```diff
@@ -366,6 +366,11 @@ public boolean checkValidAndSetMerging() {
       resetCompactionCandidateStatusForAllSourceFiles();
       return false;
     }
+    if (compactionTaskType == CompactionTaskType.NORMAL && !CompactionUtils.isDiskHaveSpace()) {
```

Comment:
Maybe we can make this judgement as a method of parent class to get rid of duplicated codes in InnerSpaceCompaction

Reply:
good idea

Reply:
good idea